src/Datadog.Trace/Propagation/CompositeTextMapPropagator.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+
+namespace Datadog.Trace.Propagation
+{
+    /// <summary>
+    /// Composite Propagator.
+    /// See more details <see href="https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/context/api-propagators.md#composite-propagator">here</see>.
+    /// </summary>
+    internal class CompositeTextMapPropagator : IPropagator
+    {
+        private readonly ICollection<IPropagator> _propagators;
```

Comment:
Could this be an `IEnumerable<IPropagator>` instead? It looks like this is only used to enumerate values and doesn't do insertions/removals

Reply:
I usually like to use ICollection when list is actually materialized - so it's quickly visible it's safe to use. With IEnumerable you are also introducing deferred execution, it can easily backfire with context like this.

Reply:
In perfect world I would use IReadOnlyCollection, but here it doesn't seem to be a pattern.

Reply:
I agree with @RassK . I used to IReadOnlyCollection for such cases